tiledb/sm/query/readers/sparse_global_order_reader.cc

```diff
@@ -346,26 +346,83 @@ SparseGlobalOrderReader<BitmapType>::create_result_tiles(
   // Get the number of fragments to process and compute per fragment memory.
   uint64_t num_fragments_to_process =
       tmp_read_state_.num_fragments_to_process();
-  per_fragment_memory_ = memory_budget_.total_budget() *
-                         memory_budget_.ratio_coords() /
-                         num_fragments_to_process;
```

Comment:
It looks like there was division by 0 happening here for the cases where `num_fragments_to_process` was found to be 0. The exception only came up after I introduced the `coordinates_budget()` method, but on rerunning the failing test I could see that we divide by zero without this method as well.

![image](https://github.com/user-attachments/assets/28988b75-3ec4-4e3c-9834-ec2b1ce648e7)